internal/querynodev2/segments/segment.go

```diff
@@ -751,6 +844,95 @@ func (s *LocalSegment) LoadFieldData(ctx context.Context, fieldID int64, rowCoun
 	return nil
 }
 
+func (s *LocalSegment) LoadDeltaData2(ctx context.Context, schema *schemapb.CollectionSchema) error {
```

Comment:
Doesn't seems to make sense to be scan in go and load delta here.

Should we move all the related logic to cpp?